xamarin/Xamarin.Forms
# AppTheme fix NRE

### Description of Change ###

Fixes an NRE (InvalidCastException) while using the OnAppTheme APIs together with xaml compilation. With this change the UI no long automatically updates when the OS theme is changed, but this way the API is at least usable. 

### Issues Resolved ### 


- fixes #

### Platforms Affected ### 


- Core/XAML (all platforms)

### Behavioral/Visual Changes ###


No more NRE when trying to use the AppTheme APIs

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###

Navigate to the Gallery app AppTheme section and notice that the app doesn't crash

### PR Checklist ###
